### PR TITLE
Backup/Restore issues

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.redhat.com/browse/ISPN-13025
https://issues.redhat.com/browse/ISPN-13026
https://issues.redhat.com/browse/ISPN-13031

### CODE REVIEW

File: commons/all/src/main/java/org/infinispan/commons/dataconversion/MediaType.java [hunk 1]
```diff
@@ -444,7 +444,7 @@ public String toStringExcludingParam(String... params) {
     */
    public boolean isBinary() {
       String customType = getClassType();
-      if (customType == null) return !this.match(MediaType.APPLICATION_OBJECT);
+      if (customType == null) return !this.withoutParameters().match(MediaType.APPLICATION_OBJECT);
```
Comment (ryanemerson) [inline commons/all/src/main/java/org/infinispan/commons/dataconversion/MediaType.java:447]: @gustavonalle I think this makes sense here, but let me know if I am missing something. Could you also take a quick look at `server/core/backup/resources/CacheResource.java ` to make sure there isn't any obvious issues with the encoding logic used when converting key/values to byte[] when creating the `CacheBackupEntry`.

File: server/core/src/test/java/org/infinispan/server/core/backup/BackupManagerImplTest.java [hunk 2]
```diff
@@ -159,6 +159,30 @@ private void invalidRestoreTest(String resourceName, BackupManager.Resources par
       );
    }
 
+   public void testBackupAndRestoreLargeCacheInteger() throws Exception {
+      String name = "testBackupAndRestore";
+      String cacheName = "cache";
+      int numEntries = 1000000;
```
[the remark this answers is not in the document]
Comment (ryanemerson) [reply server/core/src/test/java/org/infinispan/server/core/backup/BackupManagerImplTest.java:191]: The large number of entries was there due to the original issue I was trying to reproduce from the operator. The numEntries is actually not the important thing to reproduce, the issue can be reproduced with a single Cache entry if the protobuf `byte[]` length is greater than 256. I have updated the test accordingly.

**Comment by ryanemerson** [comment]
Updated and rebased

**Review by wburns — APPROVED** [review]
Changes look good to me.

File: commons/all/src/main/java/org/infinispan/commons/dataconversion/MediaType.java (hunk 1 above)
Comment (gustavonalle) [reply commons/all/src/main/java/org/infinispan/commons/dataconversion/MediaType.java:447]: ```match()```already ignores parameters, what is the test case that ```isBinary``` returns wrong results?

Comment (ryanemerson) [reply commons/all/src/main/java/org/infinispan/commons/dataconversion/MediaType.java:447]: Without this change calls to `isBinary` on MediaType `application/x-java-object;type=java.lang.String` returns true

**Comment by gustavonalle** [comment]
Merged, thanks @ryanemerson

**Pull request merged**